### ISSUE TITLE
Help with PHP-CPP

### ISSUE THREAD

**Issue by Humberto90** [CopernicaMarketingSoftware/PHP-CPP#362]
Hello,
I am interested in developing an extension for php using the PHP-CPP library. I need to implement an extension that modifies the .php files before they are sent to the apache server. Is this possible with the PHP-CPP library?
thanks for your help

**Comment by Humberto90** [comment]
The extension that I need to develop must identify encrypted files and desencriotar them before being sent to the interpreter.
Can you suggest any ideas to achieve it?
I hope you answer me,
best regards,
Humberto

**Comment by sjinks** [comment]
Yes, you set up your own hook for `zend_compile_file` during module initialization, and then in your `zend_compile_file` handle / decrypt the encrypted file and, say, pass the result to the original `zend_compile_file` or `zend_compile_string`.

PHP's ext/phar can be the source of inspiration.

**Comment by Humberto90** [comment]
Thanks for your reply. I have tried what it suggests, I managed to implement an extension using the Zen API and it has worked correctly, but I have some doubts about how to implement it using the **PHP-CPP library**. For example, to configure my hook for  `zend_compile_file ` in my extension, I need access to the data type  `zend_op_array `,

  What header should I include?

I apologize for my lack of experience, but I appreciate your cooperation.

**Comment by sjinks** [comment]
```c
#include <Zend/zend_compile.h>
```

**Comment by sjinks** [comment]
When compiling your extension, you will need to pass `$(shell php-config --includes)` to the compiler

ie:

```Makefile
%.o: %.cpp
	$(CXX) $(CPPFLAGS) `$(shell php-config --includes)` $(CXXFLAGS) -c "$<" -o "$@"
```
